### PR TITLE
Docker hub images

### LINKED ISSUE #90: Push the images to Docker Hub
See #87.

### PR DESCRIPTION
**Pull request by ash-sxn**
- fixes #90

### CODE REVIEW

**Review by gounthar — COMMENTED** [review]
Why are the images called this way _[node_agent_docker_hub_images](https://hub.docker.com/layers/ash191245141/jenkinsci-tutorials/node_agent_docker_hub_images/images/sha256-3f15a7a54917b11810fde3d721554a98fcd46b187b0c2183c7b3008059606a91?context=explore)_?

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        id: extract_branch
+      -
+        name: Build and push a simple bundled instance
+        if: contains(steps.changes.outputs.files, '00_old_one_from_proposal/Dockerfile')
+        uses: docker/build-push-action@v4
+        with:
+          context: ./00_old_one_from_proposal
+          platforms: linux/amd64, linux/aarch64
+          push: true
+          tags: ash191245141/jenkinsci-tutorials:00_old_one_${{ steps.extract_branch.outputs.branch }}
```
Comment (berviantoleo) [inline .github/workflows/docker-image.yml:66]: please use env.

Example:
```suggestion
          tags: ${{ env.DOCKERHUB_USERNAME }}/jenkinsci-tutorials:00_old_one_${{ steps.extract_branch.outputs.branch }}
```

**Comment by ash-sxn** [comment]
> Why are the images called this way node_agent_docker_hub_images?


`docker_hub_images` is the branch

**Comment by gounthar** [comment]
So how would they be called for the `main` branch?
Could we get rid of this prefix for the `main` branch?

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+          tags: ${{ env.DOCKERHUB_USERNAME }}/jenkinsci-tutorials:00_old_one_${{ steps.extract_branch.outputs.branch }}
+      -
+        name: Build and push a simple jenkins controller
+        if: contains(steps.changes.outputs.files, 'dockerfiles/Dockerfile')
+        uses: docker/build-push-action@v4
+        with:
+          context: ./dockerfiles
+          platforms: linux/amd64, linux/aarch64
+          push: true
+          tags: ${{ env.DOCKERHUB_USERNAME }}/jenkinsci-tutorials:simple_controller_${{ steps.extract_branch.outputs.branch == 'main' && '' || steps.extract_branch.outputs.branch }}
```
Comment (berviantoleo) [inline .github/workflows/docker-image.yml:78]: I prefer to use this conditional things in the steps.extract_branch. So we don't need to check in each steps.

**Review by berviantoleo — APPROVED** [review]

**Comment by gounthar** [comment]
Why is `keygen.sh` back in this PR?

**Comment by ash-sxn** [comment]
> Why is `keygen.sh` back in this PR?

previously I moved `keygen.sh` to `dockefiles/sidekick` (of which I should have made a copy)
so made a copy to the root again to work with `jenkins_init.sh`(if anybody still uses)

**Comment by gounthar** [comment]
But nobody is supposed to use `jenkins_init.sh` at this stage.

**Comment by gounthar** [comment]
I think so. 🤷 
What do @berviantoleo and @jmMeessen think of that?

**Review by gounthar — COMMENTED** [review]
Could you please give me a link to your Docker Hub repo @ash-sxn ?

**Comment by ash-sxn** [comment]
> Could you please give me a link to your Docker Hub repo @ash-sxn ?
Sorry for the late reply I didn't get the message for some reason 
anyway here's the link 
https://hub.docker.com/r/ash191245141/jenkinsci-tutorials/tags

**Comment by gounthar** [comment]
No problem, we're working asynchronously anyway. 🤷

**Comment by ash-sxn** [comment]
No, right now these images are from this branch only (suffix `docker_hub_images`) because it's not merged to the main. there are no images for the main branch yet

**Comment by ash-sxn** [comment]
should we merge this?

**Pull request merged**